### PR TITLE
[MM-37494] Check for license SKU when preparing marketplace requests

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR solves an issue where Enterprise plugins were not showing for systems with a Professional license. We are now checking the license SKU while preparing the request to be sent to the marketplace.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-37494

#### Release Note

```release-note
E20, Professional, and Enterprise license SKUs are now supported for installing Enterprise plugins.
```

### CODE REVIEW

File: model/license.go [hunk 1]
```diff
@@ -302,6 +307,13 @@ func (l *License) IsSanctionedTrial() bool {
 		(duration >= sanctionedTrialDurationLowerBound.Milliseconds() || duration <= sanctionedTrialDurationUpperBound.Milliseconds())
 }
 
+func (l *License) HasEnterprisePluginFeatures() bool {
+	return *l.Features.EnterprisePlugins ||
+		l.SkuShortName == LicenseShortSkuE20 ||
+		l.SkuShortName == LicenseShortSkuProfessional ||
```
Comment (catalintomai) [inline model/license.go:313]: Just to confirm, is "professional' an Enterprise SKU (that is, belonging to the 'Enterprise bucket)?

**Review by hanzei — APPROVED** [review]
Thanks for fixing this :+1:

File: model/license.go (hunk 1 above)
Comment (mickmister) [reply model/license.go:313]: @catalintomai In general Professional is not considered an Enterprise SKU, but Professional is above E10 in that it should have no restrictions for existing plugin features, including downloading the plugins from the marketplace. However, new individual features of plugins may discriminate by SKU on a granular level.

Please see https://community-daily.mattermost.com/core/pl/q6zkyz4hu3y1td1zfqchoca1ye for the most recent discussion on which SKUs should have access to installing enterprise plugins.

Comment (marianunez) [reply model/license.go:313]: You can also take a look at this doc for feature availability per license tiers https://docs.google.com/document/d/1nDO4NZwwxSNm_h-meJXitnNajETQ2rPam1X4D6bwfDo/edit#

**Review by marianunez — APPROVED** [review]

**Comment by amyblais** [comment]
@aaronrothschild @DHaussermann Kind reminder for PR review.

**Comment by DHaussermann** [comment]
@mickmister I have tested this locally for now as it is un-merged.
I see something I was not expecting. I only see the MS plugins when an Enterprise license is applied. Can you reproduce this?
Testing found no changes for legacy SKUs but my understanding was that for the re-packaging the plugins should also show on Professional.

Should MS Teams and Microsoft Calendar also appear with a Professional license? 
cc @aaronrothschild

**Comment by mickmister** [comment]
> I see something I was not expecting. I only see the MS plugins when an Enterprise license is applied. Can you reproduce this?

@DHaussermann I've just tested with the licenses you provided me, and I'm seeing the correct behavior:

- Starter - MS Teams **does not** show
- Professional - MS Teams **does** show
- Enterprise - MS Teams **does** show

Older licenses:

- No license - MS Teams **does not** show
- E10 - MS Teams **does not** show
- E20 - MS Teams **does** show

Let's sync up tomorrow to diagnose the issue

**Review by DHaussermann — COMMENTED** [review]
The issue I was having was deployment related. Invalid versions of the licenses causing professional to mimic starter due to an ID overlap. 
Tested and passed
_For both Cloud and On-Prem:_
Starter - MS plugins do **not** show 
Professional - MS plugins do show 
Enterprise - MS plugins do show 

_Legacy licenses:_
No license - MS Teams does not show
E10 - MS Teams does not show
E20 - MS Teams does show

LGTM!

@mickmister and @amyblais  this can now be merged and cherry-picked.

**Pull request merged**